misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/DependencySourceResolver.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ballerinalang.debugadapter.utils;
+
+import com.sun.jdi.Location;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleName;
+import io.ballerina.projects.PackageName;
+import io.ballerina.projects.PackageResolution;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ResolvedPackageDependency;
+import io.ballerina.projects.util.ProjectUtils;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Concrete implementation of Ballerina package dependency (stdlib, central modules) source resolver.
+ *
+ * @since 2.0.0
+ */
+public class DependencySourceResolver extends SourceResolver {
+
+    DependencySourceResolver(Project sourceProject) {
+        super(sourceProject);
+    }
+
+    @Override
+    public boolean canResolve(Location location) {
+        try {
+            LocationInfo locationInfo = new LocationInfo(location);
+            if (!locationInfo.isValid()) {
+                return false;
+            }
+
+            List<PackageName> possiblePackageNames = ProjectUtils.getPossiblePackageNames(locationInfo.moduleName());
```

Comment:
Better to avoid using `ProjectUtils.getPossiblePackageNames`. The dependency graph will only contain one module out of all possibilities. Can't we directly iterate the nodes in the dependency graph and check the org and the module name?
Refer: https://github.com/ballerina-platform/ballerina-lang/blob/master/project-api/project-api-test/src/test/java/io/ballerina/projects/test/DependencyFilePathsTest.java#L85

Reply:
@azinneera Refactored and removed `ProjectUtils` usages in 91aeb42.